src/openfermion/measurements/_probability_distributions.py

```diff
@@ -0,0 +1,358 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""
+Classes to store probability distributions for estimation
+of random variables.
+"""
+import numpy
+from scipy import sparse
+from numpy import cos, sin, pi
+
+
+class FourierProbabilityDist(object):
+    """
+    Stores a multivariant Fourier representation of a periodic function:
+
+    f(phi_0,phi_1,...) = sum_j A_j f_j
+    f_j = sum_n (c_{j,2n} cos(n*phi_j) + c_{j,2n-1} sin(n*phi_j))
+
+    In particular, this class stores values of A_j, c_{j,2n},
+    c_{j,2n+1} from the above equation.
+
+    It also provides routines for multiplying f by functions of the form
+    cos^2(n*phi_j + beta), and for calculating maximum-likelihood
+    distributions of the A_j variables.
+
+    More details can be found in arXiv:1809.09697, Appendix C.
+    """
+
+    def __init__(self,
+                 init_amplitude_guess=None,
+                 init_amplitude_vars=None,
+                 num_vectors=1,
+                 num_freqs=1000,
+                 vector_guess=None):
+        """
+        Args:
+            init_amplitude_guess (numpy array or list):
+                We take an initial normal distribution for the amplitudes
+                which requires for each a mean and a variance - 'guess' is
+                the mean
+            init_amplitude_vars (numpy array or list): variance of the
+                estimated distribution in init_amplitude_guess
+            num_vectors (int): number of phases phi_j to estimate
+            num_freqs (int): number of wave components cos(n*phi_j)
+                to store coefficients for (this provides a cut-off to
+                the sensitivity of the function).
+            max_n (int): the maximum n required by the user in
+                multiplying the stored function functions of the form
+                cos(n*phi_j + beta).
+            vector_guess (numpy array): a prior estimate of the function,
```

Comment:
What function is this a prior estimate of? The function being represented by this class? That seems circular to me.

Reply:
Yes, that's correct. You need a prior for a Bayesian estimator, which you can get from other means (e.g. classical approximations, or taking something generic).

Reply:
Hmm I thought we agreed to separate the estimator concept from the probability distribution concept. We need to avoid this circular logic in the description of this variable.

Reply:
Fair point. I've edited the docstring to remove reference to this and updated the variable names to make more sense, hope this looks better now.